### PR TITLE
Throw errors if result of `startUri` has errors

### PR DESCRIPTION
**Pull request by dpgraham**
* I need this for another PR because starters was unsuccessfully starting activities but was still returning zero as the error code so I need a way to check that 

(related PR: https://github.com/appium/appium-uiautomator2-driver/pull/151)

### CODE REVIEW

File: lib/tools/apk-utils.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ apkUtilsMethods.startUri = async function (uri, pkg) {
   try {
     let args = ["am", "start", "-W", "-a", "android.intent.action.VIEW", "-d",
                 uri.replace(/&/g, '\\&'), pkg];
-    await this.shell(args);
+    return await this.shell(args);
```
Comment (mykola-mokhnach) [inline lib/tools/apk-utils.js]: please  update the docstring as well

**Comment by mykola-mokhnach** [comment]
Would it be possible to add failures verification into this method or you'd prefer to do this in the caller class?

**Comment by dpgraham** [comment]
@mykola-mokhnach I updated `startUri` to catch the failures. I think it probably doesn't hurt to put in here instead of `appium-uiautomator2-driver`.

**Comment by dpgraham** [comment]
@mykola-mokhnach  Done.

File: test/unit/apk-utils-specs.js [hunk 2]
```diff
@@ -465,14 +465,32 @@ describe('Apk-utils', function () {
     });
   }));
   describe('startUri', withMocks({adb}, (mocks) => {
+    before(async function () {
```
Comment (mykola-mokhnach) [inline test/unit/apk-utils-specs.js]: do we need this empty func?

Comment (dpgraham) [reply test/unit/apk-utils-specs.js]: No, it's gone now.

**Comment by dpgraham** [comment]
@mykola-mokhnach When the tests pass, am I good to merge then publish?

**Pull request merged**